x-pack/platform/plugins/shared/embeddable_alerts_table/public/components/config_editor_flyout.tsx

```diff
@@ -0,0 +1,242 @@
+  const [filters, setFilters] = useState<
+    EmbeddableAlertsTableConfig['query']['filters'] | undefined
+  >(initialConfig?.query?.filters ?? EMPTY_FILTERS);
+  const formRef = useRef<AlertsFiltersFormHandle>(null);
+  const {
+    data: ruleTypes,
+    isLoading: isLoadingRuleTypes,
+    isError: cannotLoadRuleTypes,
+  } = useGetInternalRuleTypesQuery({ http });
+  const availableSolutions = useMemo(
+    () => (ruleTypes ? getAvailableSolutions(ruleTypes) : undefined),
+    [ruleTypes]
+  );
+  const [solution, setSolution] = useState<EmbeddableAlertsTableConfig['solution'] | undefined>(
+    initialConfig?.solution ??
+      (availableSolutions && availableSolutions.length === 1 ? availableSolutions[0] : undefined)
+  );
+  const ruleTypeIds = useMemo(
+    () => (!ruleTypes || !solution ? [] : getRuleTypeIdsForSolution(ruleTypes, solution)),
+    [ruleTypes, solution]
+  );
+
+  const resetFilters = useCallback(() => {
+    setFilters(EMPTY_FILTERS);
+  }, []);
+
+  const onSolutionChange = useCallback(
+    (newSolution: RuleTypeSolution) => {
+      if (
+        solution != null &&
+        newSolution !== solution &&
+        filters != null &&
+        !isEmptyExpression(filters)
+      ) {
+        // Filters are incompatible between different solutions and thus must be reset
+        // Ask for confirmation before doing so
+        overlays
+          .openConfirm(SWITCH_SOLUTION_CONFIRM_MESSAGE, {
+            title: SWITCH_SOLUTION_CONFIRM_TITLE,
+            buttonColor: 'danger',
+            cancelButtonText: SWITCH_SOLUTION_CONFIRM_CANCEL_MESSAGE,
+            confirmButtonText: SWITCH_SOLUTION_CONFIRM_CONFIRM_MESSAGE,
+          })
+          .then((result) => {
+            if (result) {
+              // User accepted to switch solution
+              setSolution(newSolution);
+              resetFilters();
+            }
+          });
+      } else {
+        setSolution(newSolution);
+      }
+    },
+    [filters, overlays, resetFilters, solution]
+  );
+
+  useEffectOnce(() => {
+    if (!solution && flyoutBodyRef.current) {
+      const panel = flyoutBodyRef.current.closest('.euiFlyout');
+      // Waiting for the flyuot entry animation to finish before auto-opening the solution
+      // selector, otherwise the options popover cannot position itself correctly
+      panel?.addEventListener('animationend', () => solutionSelectorRef.current?.openPopover(), {
+        once: true,
+      });
+      return () => panel?.removeEventListener('animationend', focus);
+    }
+  });
+
+  return (
+    <>
+      <EuiFlyoutHeader hasBorder={true}>
+        <EuiTitle size="s">
+          <h2>{!initialConfig ? CONFIG_EDITOR_ADD_TABLE_TITLE : CONFIG_EDITOR_EDIT_TABLE_TITLE}</h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+
+      <EuiFlyoutBody>
+        <EuiFlexGroup direction="column" gutterSize="m" ref={flyoutBodyRef}>
+          <EuiFlexItem>
+            {availableSolutions && availableSolutions.length > 1 && (
```

Comment:
Extracted the auto-hide logic from inside the `AlertsSolutionSelector` component